p2p/peers.go

```diff
@@ -40,8 +41,24 @@ func (pi PeersImpl) Close() {
 
 func (pi PeersImpl) GetPeers() []Peer {
 	peers := pi.snapshot.Load().([]Peer)
-	pi.With().Info("now connected", log.Int("n_peers", len(peers)))
-	return peers
+	cpy := make([]Peer, len(peers))
```

Comment:
why do you copy? isn't snapshot loading always returns a copy? Until now we would've just returned peers.

Reply:
because its a slice it returns a pointer so the copy is necessary